### ISSUE TITLE
@debounce is named improperly — should be @throttle

### ISSUE THREAD

**Issue by ivancuric** [developit/decko#9]
In the [docs](https://github.com/developit/decko#debounce) its says:
> To debounce means "call this at most once per N ms". All outward function calls get collated into a single inward call, and only the latest (most recent) arguments as passed on to the debounced function.

What you're doing is throttling, not debouncing.
[lodash.throttle](https://lodash.com/docs/4.17.4#throttle):
> Creates a throttled function that only invokes `func` at most once per every `wait` milliseconds. 

[lodash.debounce](https://lodash.com/docs/4.17.4#debounce):
> Creates a debounced function that delays invoking `func` until after `wait` milliseconds have elapsed since the last time the debounced function was invoked.

**Comment by developit** [comment]
Agreed! It's a difficult thing to change now that its there, but what about adding an option and making the default an actual debounce?  @pl12133 called this out to me a couple weeks ago, I'd forgotten about the name mixup.

**Comment by ivancuric** [comment]
The sooner you do it the better if you see a future for this package. We have enough misnamed "oh-it's-too-late-now" stuff in the web dev industry as it is :)

Anyway, decided to use https://www.npmjs.com/package/core-decorators instead for the project I'm working on since I do need both. The implementations are quite similar as well.

**Comment by developit** [comment]
Oh nice, I hadn't seen that module! Looks good.

**Comment by Connorelsea** [comment]
I'd love to have a real debounce option here somehow. Separating throttle and debounce would make a lot of sense. Also core-decorators seems to have deprecated their debounce and throttle decorators for some reason
